tasks/mpi/tsatsyn_a_increasing_contrast_by_histogram/src/ops_mpi.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Nesterov Alexander
+#include "mpi/tsatsyn_a_increasing_contrast_by_histogram/include/ops_mpi.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <ctime>
+#include <functional>
+#include <random>
+#include <string>
+#include <thread>
+#include <vector>
+
+bool tsatsyn_a_increasing_contrast_by_histogram_mpi::TestMPITaskParallel::validation() {
+  internal_order_test();
+  if (world.rank() == 0) {
+    return (taskData->outputs_count[0] == 1 && taskData->inputs_count[0] > 0);
+  }
+  return true;
+}
+bool tsatsyn_a_increasing_contrast_by_histogram_mpi::TestMPITaskParallel::pre_processing() {
+  internal_order_test();
+  if (world.rank() == 0) {
+    input_data.resize(taskData->inputs_count[0]);
+    auto* tempPtr = reinterpret_cast<int*>(taskData->inputs[0]);
+    std::copy(tempPtr, tempPtr + taskData->inputs_count[0], input_data.begin());
+  }
+
+  return true;
+}
+bool tsatsyn_a_increasing_contrast_by_histogram_mpi::TestMPITaskParallel::run() {
+  internal_order_test();
+  std::vector<double> local_data;
+  int width;
+  int height;
+  if (world.rank() == 0) {
+    auto* tempPtr = reinterpret_cast<int*>(taskData->inputs[1]);
+    width = tempPtr[0];
+    height = tempPtr[1];
+
+    for (int proc = 1; proc < world.size(); proc++) {
+      local_data.clear();
+      for (int i = proc; i < static_cast<int>(input_data.size()); i += world.size()) {
+        local_data.emplace_back(input_data[i]);
```

Comment:
I suggest to use this instead of .emplace_back(input_data[i])
```suggestion
        local_data.insert(local_data.end(), input_data.begin() + i, input_data.end());
```

Reply:
даже не совсем так, если бы шаг в цикле был i++ то да, но там шаг i+=world.size()